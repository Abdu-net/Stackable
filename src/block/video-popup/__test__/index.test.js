import { edit, save } from '../index'

describe( 'Video Popup', () => {
	test( 'block edit matches snapshot', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				videoLink: 'QJ9ygdD2sIY',
				mediaLink: 'https://images.unsplash.com/photo-1506269351850-0428eaed2193?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=a1512d7659e4817df8217cdb9aa09d7a&auto=format&fit=crop&w=1050&q=80',
				mediaID: '',
				overlayColor: '#000000',
				playButtonType: 'normal',
			},
		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshot', () => {
		const wrapper = save( {
			attributes: {
				videoLink: 'QJ9ygdD2sIY',
				mediaLink: 'https://images.unsplash.com/photo-1506269351850-0428eaed2193?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=a1512d7659e4817df8217cdb9aa09d7a&auto=format&fit=crop&w=1050&q=80',
				mediaID: '',
				overlayColor: '#000000',
				playButtonType: 'normal',
			},
		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
