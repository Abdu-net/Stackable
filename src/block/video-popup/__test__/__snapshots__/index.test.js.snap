// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Popup block saved block HTML shouldn't change 1`] = `
<div
  className="ugb-video-popup ugb-video-popup--v2 ugb-video-popup--design-design ugb-video-popup--button-outline ugb--background-opacity-2 ugb--has-background ugb--has-background-image ugb--shadow-1"
  data-video="videoID"
  style={
    Object {
      "--ugb-background-color": "#123456",
      "backgroundColor": "#123456",
      "backgroundImage": "url(backgroundImageURL)",
      "borderRadius": 6,
    }
  }
>
  <div
    className="ugb-video-popup__wrapper"
  >
    <a
      className="ugb-video-popup__overlay"
      href="#"
    />
    <span
      className="ugb-video-popup__play-button"
    >
      <div />
    </span>
  </div>
</div>
`;

exports[`Video Popup block saved default block HTML shouldn't change 1`] = `
<div
  className="ugb-video-popup ugb-video-popup--v2 ugb-video-popup--design-basic ugb-video-popup--button-normal ugb--background-opacity-5 ugb--has-background"
  style={
    Object {
      "--ugb-background-color": undefined,
      "backgroundColor": "#000000",
      "backgroundImage": undefined,
      "borderRadius": undefined,
    }
  }
>
  <div
    className="ugb-video-popup__wrapper"
  >
    <a
      className="ugb-video-popup__overlay"
      href="#"
    />
    <span
      className="ugb-video-popup__play-button"
    >
      <div />
    </span>
  </div>
</div>
`;
